lib/internal/navigator.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const {
+  ObjectDefineProperties,
+  ReflectConstruct,
+} = primordials;
+
+const {
+  ERR_ILLEGAL_CONSTRUCTOR,
+} = require('internal/errors').codes;
+
+const {
+  kEnumerableProperty,
+} = require('internal/util');
+
+const {
+  getAvailableParallelism,
+} = internalBinding('os');
+
+class Navigator {
+  constructor() {
+    throw ERR_ILLEGAL_CONSTRUCTOR();
+  }
+
+  /**
+   * @return {number}
+   */
+  get hardwareConcurrency() {
```

Comment:
needs a brand check, otherwise something like this won't throw
```js
Object.getOwnPropertyDescriptor(Navigator.prototype, 'hardwareConcurrency').get()
```

Reply:
Used private properties to avoid checking for brand validation.